lib/beaker-pe/install/pe_utils.rb

```diff
@@ -1549,6 +1591,34 @@ def download_pe_conf_if_master(host)
             end
           end
         end
+
+        # Method to install just the agent nodes
+        # This method can be called only after installing PE on all other nodes
+        # including infrastructure nodes, loadbalancer, hubs and spokes
+        # PE-22051
+        # @param [Array] agent only nodes from Beaker hosts
+        # @param [Hash] opts The Beaker options hash
+        #
+        def install_agents_only_on(agentnodes, opts)
```

Comment:
I am inclined to add that change as a different PR after this goes in since that will be a change affecting almost all of the CI jobs and if there is a failure it will be easy to trace back. What do you think?

Reply:
As we discussed over chat, I have added these changes. Now simple_monolithic_install() will call install_agents_only_on() for installing agents.